### DISCUSSION TITLE
Remove the pop mode from PushConsumer

### DISCUSSION THREAD

**Discussion by drpmma** [apache/rocketmq#6876] [General]
## Summary
The semantics of the pop mode does not align with PushConsumer, and there are issues with clearing timeout message, rebalancing, and broadcasting. Additionally, implementing the pop mode increases the complexity of the client. Therefore, it is recommended to remove the pop mode from PushConsumer.

## Motivation
It's necessary to keep the core of rocketmq client simple where the pop mode introduces complexity and disrupts the original PushConsumer model.

**Comment by aaron-ai** [comment]
+1

**Comment by chenzlalvin** [comment]
Keep client simple, can't agree more!

**Comment by lollipopjin** [comment]
+1
